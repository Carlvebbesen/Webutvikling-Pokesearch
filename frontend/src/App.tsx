import React from 'react';
import {Redirect, Route, Switch} from "react-router"
import {BrowserRouter} from 'react-router-dom'
import OverviewPage from "./pages/Overview";
import Test from "./components/Test"
import "./index.css"
import {RecoilRoot} from "recoil";

function App() {
    return (
        <div className="App">
            <RecoilRoot>
            <BrowserRouter>
                <Switch>
                    <Route exact path={"/prosjekt3"}>
                        <OverviewPage/>
                    </Route>
                    <Route exact path={"/test"}>
                        <div>
                            <Test/>
                        </div>
                    </Route>
                    <Redirect to={"/prosjekt3"}/>
                </Switch>
            </BrowserRouter>
            </RecoilRoot>
        </div>
    );
}

export default App;
