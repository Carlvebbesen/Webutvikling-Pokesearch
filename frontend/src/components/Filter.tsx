
import React from "react"
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import TextField from "@mui/material/TextField";

interface iFilter{
    name: string
}
{/*
const Filter FC<iFilter>= (props)=>{
    return(
        <div>
            <p>hei</p>
            <FormControl sx={{ m: 1, minWidth: 120 }}>
                <InputLabel id="demo-simple-select-helper-label">City</InputLabel>
                <Select
                    labelId="demo-simple-select-helper-label"
                    id="demo-simple-select-helper"
                    value={props.city}
                    label="City"
                    onChange={(event) => props.setCity(event.target.value)}
                >
                    <MenuItem value="All">
                        <em>None</em>
                    </MenuItem>
                    <MenuItem value={"Oslo"}>Oslo</MenuItem>
                    <MenuItem value={"Trondheim"}>Trondheim</MenuItem>
                    <MenuItem value={"Bergen"}>Bergen</MenuItem>
                    <MenuItem value={"Tromsø"}>Tromsø</MenuItem>
                    <MenuItem value={"Stavanger"}>Stavanger</MenuItem>
                </Select>
            </FormControl>
            <TextField
                id="filled-number"
                label="Maxiumum price"
                type="number"
                InputLabelProps={{
                    shrink: true,
                }}
                value={props.price}
                onChange={(event) =>
                    props.setPrice(event.target.value as unknown as number)
                }
                variant="filled"
            />
        </div>
    )
}

export default Filter
*/}