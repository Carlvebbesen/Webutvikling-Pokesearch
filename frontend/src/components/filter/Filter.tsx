import React, {FC} from "react"
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select, {SelectChangeEvent} from "@mui/material/Select";
import TextField from "@mui/material/TextField";
import {PokemonTypes} from "../../utils/Values";
import {OutlinedInput, Rating} from "@mui/material";
import Box from '@mui/material/Box';
import Chip from '@mui/material/Chip';
import {useTheme, Theme} from "@mui/material/styles";
import style from "./Filter.module.css"
import { getPokeTypeIcon } from "../../static/typeIcons/pokeTypeIcons";

function getStyles(name: string, selected: readonly string[], theme: Theme) {
    return {
        fontWeight:
            selected.indexOf(name) === -1
                ? theme.typography.fontWeightRegular
                : theme.typography.fontWeightMedium,
        display: "flex"
    };
}

interface iFilter {
    type: string[]
    name: string
    rating: number
    setType: Function
    setName: Function
    setRating: Function
}


const Filter: FC<iFilter> = (props) => {

    const theme = useTheme();

    const ITEM_HEIGHT = 48;
    const ITEM_PADDING_TOP = 8;
    const MenuProps = {
        PaperProps: {
            style: {
                maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
                width: 250,
            },
        },
    };


    const handleChange = (event: SelectChangeEvent<typeof props.type>) => {
        const value = event.target.value;
        if (value.length <= 2) {
            props.setType(
                // On autofill we get a the stringified value.
                typeof value === 'string' ? value.split(',') : value
            );
        } else {
            props.setType(
                typeof value === 'string' ? value.split(',') : value.splice(1,value.length)
            )
        }
    };


    return (
        <div className={style.filter}>
            <TextField sx={{m: 1, width: 200}}
                       id="outlined-search"
                       label="Name"
                       type="search"
                       value={props.name}
                       onChange={event =>
                           props.setName(event.target.value)}
                       data-cy="name_input"
            />
            <FormControl sx={{m: 1, width: 200}} data-cy="type-selector-container">
                <InputLabel id="demo-multiple-chip-label">Type (Max 2)</InputLabel>
                <Select
                    data-cy="type-selector"
                    labelId="multiple-chip-label"
                    id="Type"
                    multiple
                    value={props.type}
                    onChange={handleChange}
                    input={<OutlinedInput id="select-multiple-chip" label="Type"/>}
                    renderValue={(selected) => (
                        <Box sx={{display: 'flex', flexWrap: 'wrap', gap: 0.5}}>
                            {selected.map((value) => (
                                <Chip key={value} label={value}/>
                            ))}
                        </Box>
                    )}
                    MenuProps={MenuProps}
                >
                    {PokemonTypes.map((type) => (
                        <MenuItem sx={{display: "block"}}
                                  key={type}
                                  value={type}
                                  style={getStyles(type, props.type, theme)}
                                  data-cy={`type-option-${type}`}
                        >
                            <><img title={type} style={{marginRight: "10px"}} height="20" alt="pokemonTypes" src={getPokeTypeIcon(type)}/>{type} </>
                        </MenuItem>
                    ))}
                </Select>
            </FormControl>
            <div>
                <p>Minimum rating:</p>
                <div onClick={()=> props.setRating(0)}>
                <Rating
                    name="simple-controlled"
                    value={props.rating}
                    precision={0.5}
                    onChange={(event, star) => {
                        if (star != null) {
                            props.setRating(star);
                        }
                        else {
                            props.setRating(0);
                        }
                    }}
                    />
                    </div>
            </div>

        </div>
    )
}

export default Filter