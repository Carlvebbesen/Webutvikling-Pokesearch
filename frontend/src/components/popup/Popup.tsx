import React, {FC, useEffect, useState} from "react"
import style from "./Popup.module.css"
import {useMutation, useQuery} from "@apollo/client";
import {ADD_RATING_BY_POKEMONID, GET_POKEMON_BY_ID} from "../../queries";
import CircularProgress from '@mui/material/CircularProgress';
import {capitalize, Rating} from "@mui/material";
import SendIcon from '@mui/icons-material/Send';
import {getPokeTypeIcon} from "../../static/typeIcons/pokeTypeIcons";
import { StatTable } from "../statTable/statTable";
import { BsXSquare } from 'react-icons/bs';
import { toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import Team from "../team/Team";
import { useResetRecoilState } from "recoil";

interface iPopup {
    pokemonId: number,
    setOpen: (id: number|null)=> void,
}


const Popup: FC<iPopup> = ({pokemonId, setOpen}) => {
    const [rating, setRating] = useState<number>(0)
    const [disable, setDisable] = useState<boolean>(false)
    const {data, error, loading} = useQuery(GET_POKEMON_BY_ID, {
        variables: {input: {id: pokemonId}}
    });
    const [totalRating, setTotalRating] = useState<number| null>(null);
    const [mutateFunction] = useMutation(ADD_RATING_BY_POKEMONID);

    useEffect(()=>{
        const previousRating = localStorage.getItem(pokemonId.toString());
        if(previousRating!== null) {
            setRating(parseInt(previousRating));
            setDisable(true);
        }
    },[]);


    const handleRating = () => {
        setDisable(true);
        localStorage.setItem(pokemonId.toString(), rating.toString());
        mutateFunction({variables: {input:{
            id: pokemonId,
            rating: rating,
        }}}).then((response) => {
            setTotalRating(response.data.ratePokemon.rating)
            toast.success("Rating submitted");
        });
    }

    return (
        <div id="inner" className={style.popupInner}>
            {loading || error ? <CircularProgress/>
                :<>
                <div className={style.wrapper}>

                <div className={style.container}>
                    <div
                        className={style.close}>
                        <BsXSquare onClick={() => {
                            setOpen(null)
                        }}/>
                    </div>
                    <div className={style.spriteSection}>
                        <h2>{capitalize(data?.getPokemonById.name)}</h2>
                        <img
                            className={style.spritePic}
                            src={data?.getPokemonById.sprite_url}
                            alt={"Picture of " + data?.getPokemonById.name}/>
                        <div className={style.dataEntry}>
                            <span>Dex number</span>
                            <span>{pokemonId}</span>
                        </div>
                        <div className={style.dataEntry}>
                            <span>Typing</span>
                            <div>
                                {data?.getPokemonById.pokeTypes.map((type: string) =>
                                    <img
                                    style={{marginRight: "10px"}}
                                    height="20"
                                    src={getPokeTypeIcon(type)}
                                    alt={type}/>)}
                            </div>
                        </div>
                        <div className={style.dataEntry}>
                            <span>Weight</span>
                            <span>{data?.getPokemonById.weight} kg</span>
                        </div>
                        <div className={style.dataEntry}>
                            <span>Average rating</span>
                            <Rating
                                name="read-only"
                                defaultValue={0}
                                precision={0.1}
                                value={data?.getPokemonById.rating}
                                readOnly
                                size="small"/>
                        </div>
                        <div className={style.dataEntry}>
                            <span>Usage percentage</span>
                            <span>{data?.getPokemonById.usage_percentage}%</span>
                        </div>
                    </div>
                    <div className={style.dataSection}>
                        <StatTable stats={[
                            {name: 'hp', value: data?.getPokemonById.stats.hp},
                            {name: 'attack', value: data?.getPokemonById.stats.attack},
                            {name: 'defense', value: data?.getPokemonById.stats.defense},
                            {name: 'special-attack', value: data?.getPokemonById.stats.special_attack},
                            {name: 'sepcial-defense', value: data?.getPokemonById.stats.special_defense},
                            {name: 'speed', value: data?.getPokemonById.stats.speed},
                        ]} />
                        <div>
                            <h5>Give rating</h5>
                            <Rating
                                name="simple-controlled"
                                value={rating}
                                precision={0.5}
                                onChange={(event, newValue) => {
                                    setRating(newValue ? newValue : 0)
                                }}
                                disabled={disable}
                                />
                            <button className={style.rating} onClick={handleRating} disabled={(rating === 0) || disable}><SendIcon/></button>
                        </div>
                    </div>
                </div>
                <Team currentPokemon={data?.getPokemonById}/>   
                </div>
                </>
                }
        </div>
    );
}

export default Popup