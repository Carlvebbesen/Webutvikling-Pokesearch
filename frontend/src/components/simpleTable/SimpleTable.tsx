import {Pokemon} from "../../types/Pokemon";
import React, {FC, useState} from "react";
import style from "./SimpleTable.module.css";
import Rating from "@mui/material/Rating";
import Team from "../team/Team";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import IconButton from "@material-ui/core/IconButton";
import KeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import KeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import Paper from "@material-ui/core/Paper";
import Table from "@material-ui/core/Table";
import TableHead from "@material-ui/core/TableHead";
import TableBody from "@material-ui/core/TableBody";
import TablePagination from "@mui/material/TablePagination";
import {capitalize} from "@mui/material";


interface iDetails extends Pokemon {
    setRating: Function;
    getRating: Function;
}

const Details: FC<iDetails> = (props) => {
    const teamMembers = [1, 2, 3, 4]
    return (
        <div className={style.outerWrapper}>
            <h1>{capitalize(props.name)}</h1>
            <div className={style.wrapper}>
                <div className={style.detailsDiv}>
                    <img src={props.sprite_url} className={style.bigpic} alt="pokemon"/>
                </div>
                <div className={style.detailsDiv}>
                    <p>type: {props.type}</p>
                    <p>number: {props.id}</p>
                    <p>weight: {props.weight}</p>
                </div>
                <div className={style.detailsDiv}>
                    <div>
                        <p>average rating of {props.number_of_ratings} people</p>
                        <Rating name="read-only" defaultValue={0} precision={0.1} value={props.rating} readOnly/>
                    </div>
                    <p>Used by: {props.usage_percentage * 100}% of teams</p>
                    <div>
                        <p>your rating</p>
                        <Rating
                            name="simple-controlled"
                            value={props.getRating(props.id)}
                            precision={0.5}
                            onChange={(event, newValue) => {
                                props.setRating(props.id, newValue)
                            }}
                        />
                    </div>
                </div>
                <div className={style.detailsDiv}>
                    <Team members={teamMembers} pokemonid={props.id}/>
                </div>
            </div>
        </div>
    )
}

const ExpandableTableRow = ({children, expandComponent, ...otherProps}: any) => {
    const [isExpanded, setIsExpanded] = useState(false);

    return (
        <>
            <TableRow {...otherProps}>
                <TableCell padding="checkbox">
                    <IconButton onClick={() => setIsExpanded(!isExpanded)}>
                        {isExpanded ? <KeyboardArrowUpIcon/> : <KeyboardArrowDownIcon/>}
                    </IconButton>
                </TableCell>
                {children}
            </TableRow>
            {isExpanded && (
                <TableRow className={style.expanding}>
                    <TableCell padding="checkbox"/>
                    {expandComponent}
                </TableRow>
            )}
        </>
    );
};

interface iSimpleTable {
    rows: Pokemon[]
    map: Map<number, number>
    page: number
    rowsPerPage: number
    setMap: Function
    setPage: Function
    setRowsPerPage: Function
}

const SimpleTable: FC<iSimpleTable> = (props) => {
    const handleChangePage = (event: unknown, newPage: number) => {
        props.setPage(newPage);
    };

    const upsert = (key: number, value: number) => {
        props.setMap((prev: Map<number, number>) => new Map(prev).set(key, value));
    }

    const getStars = (key: number) => {
        if (props.map.has(key)) {
            return props.map.get(key)
        } else {
            return 0
        }
    }

    const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
        props.setRowsPerPage(+event.target.value);
        props.setPage(0);
    };

    return (
        <div>
            <Paper className={style.root}>
                <Table stickyHeader className={style.table} aria-label="sticky table">
                    <TableHead>
                        <TableRow>
                            <TableCell padding="checkbox"/>
                            <TableCell>Name</TableCell>
                            <TableCell align="right">HP</TableCell>
                            <TableCell align="right">Attack</TableCell>
                            <TableCell align="right">Defence</TableCell>
                            <TableCell align="right">Sp. Atk</TableCell>
                            <TableCell align="right">Sp. Def</TableCell>
                            <TableCell align="right">Speed</TableCell>
                            <TableCell align="right">Total</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {props.rows.slice(
                            props.page * props.rowsPerPage, props.page * props.rowsPerPage + props.rowsPerPage)
                            .map(row => (
                                <ExpandableTableRow
                                    key={capitalize(row.name)}
                                    expandComponent={<TableCell colSpan={8}>
                                        {<Details
                                            getRating={getStars}
                                            setRating={upsert}
                                            id={row.id}
                                            name={row.name}
                                            type={row.type}
                                            stats={row.stats}
                                            weight={row.weight}
                                            rating={row.rating}
                                            number_of_ratings={10}
                                            usage_percentage={row.usage_percentage}
                                            sprite_url={row.sprite_url}
                                        />
                                        }
                                    </TableCell>}
                                >
                                    <TableCell component="th" scope="row">
                                        <img src={row.sprite_url} alt="pokemon"></img>{capitalize(row.name)}
                                    </TableCell>
                                    <TableCell align="right">{row.stats.find(e => e.name === "hp")?.value}</TableCell>
                                    <TableCell
                                        align="right">{row.stats.find(e => e.name === "attack")?.value}</TableCell>
                                    <TableCell
                                        align="right">{row.stats.find(e => e.name === "defence")?.value}</TableCell>
                                    <TableCell
                                        align="right">{row.stats.find(e => e.name === "sp.atk")?.value}</TableCell>
                                    <TableCell
                                        align="right">{row.stats.find(e => e.name === "sp.def")?.value}</TableCell>
                                    <TableCell
                                        align="right">{row.stats.find(e => e.name === "speed")?.value}</TableCell>
                                    <TableCell
                                        align="right">{row.stats.map(e => e.value as number).reduce((a, b) => a + b, 0)}</TableCell>
                                </ExpandableTableRow>
                            ))
                        }
                    </TableBody>
                </Table>
                <TablePagination //TODO: https://mui.com/api/table-pagination/
                    rowsPerPageOptions={[5, 10, 15]}
                    count={props.rows.length}
                    page={props.page}
                    rowsPerPage={props.rowsPerPage}
                    onPageChange={handleChangePage}
                    onRowsPerPageChange={handleChangeRowsPerPage}
                />
            </Paper>
        </div>
    )
        ;
}

export default SimpleTable