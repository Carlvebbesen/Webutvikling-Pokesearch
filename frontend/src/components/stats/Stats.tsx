import React, {FC, useRef} from "react"
import style from "./Stats.module.css"

interface iSingleStat {
    name: string,
    width: number
    color: string
    value: number
}


const SingleStat: FC<iSingleStat> = (props) => {
    return (
        <div className={style.singleStat}>
            <div className={style.nameVal}>
                <p>{props.name}</p>
                <p>{props.value}</p>
            </div>
            <div className={style.nameStat}>
            <div style={{width: props.width + "%", backgroundColor: props.color, height: "2em", alignContent:"center"}}>
            </div>
            </div>
        </div>
    )
}

interface iStats {
    Hp: number,
    Atk: number,
    Def: number,
    SpAtk: number,
    SpDef: number,
    Speed: number
}

const Stats: FC<iStats> = (props) => {
    const max: number = Math.max(props.Hp, props.Atk, props.Def, props.SpAtk, props.SpDef, props.Speed)
    const ref = useRef(null);


    return (
        <div className={style.outer} ref={ref}>
            <SingleStat color={"red"} name={"HP"} width={(props.Hp / max) * 100} value={props.Hp}/>
            <SingleStat color={"orange"} name={"Atk"} width={(props.Atk / max) * 100} value={props.Atk}/>
            <SingleStat color={"yellow"} name={"Def"} width={(props.Def / max) * 100} value={props.Def}/>
            <SingleStat color={"blue"} name={"SpAtk"} width={(props.SpAtk / max) * 100} value={props.SpAtk}/>
            <SingleStat color={"green"} name={"SpDef"} width={(props.SpDef / max) * 100} value={props.SpDef}/>
            <SingleStat color={"purple"} name={"Speed"} width={(props.Speed / max) * 100} value={props.Speed}/>
        </div>
    )
}


export default Stats