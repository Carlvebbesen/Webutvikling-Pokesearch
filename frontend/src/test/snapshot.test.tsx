import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { MockedProvider } from '@apollo/client/testing';
import App from '../App';
import Backdrop from '../components/backdrop/backdrop';
import Filter from '../components/filter/Filter';
import Popup from '../components/popup/Popup';
import SimpleTable from '../components/simpleTable/SimpleTable';
import { mocks, filteredData } from './testData';
import SortingButtonsList from '../components/sort/sortingButtonsList';
import SortingButton from '../components/sort/sortingButton';
import Stats from '../components/stats/Stats';
import TableListRow from '../components/tableListRow/tableListRow';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

it('Snapshot test for App component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
it('Snapshot test for Backdrop component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Backdrop clicked={()=>{}} show={true}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
it('Snapshot test for Filter component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Filter type={[]} name={''} rating={0} setType={()=>{}} setName={()=>{}} setRating={()=>{}}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
it('Snapshot test for PopUp component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Popup pokemonID={1} setOpen={()=> {}}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
  it('Snapshot test for SimpleTable component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <SimpleTable data={filteredData} changePage={()=>{}} changeRowsPerPage={()=>{}} setPopUpId={()=>{}} page={0} rowsPerPage={25} activeSortButton={"hp"} sortPokemon={()=>{}}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
  it('Snapshot test for SortList component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <SortingButtonsList sortByValue={()=>{}} activeButton={"HP"}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
  it('Snapshot test for SortButton component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <SortingButton name={'HP'} currentSort={undefined} sort={()=>{}}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
  it('Snapshot test for Stats component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Stats Hp={50} Atk={50} Def={50} SpAtk={50} SpDef={50} Speed={50}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });
  
  it('Snapshot test for ListPokemonRow component', () => {
    const component = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <TableListRow pokemon={filteredData.pokemons[0]} setPopUpShow={()=>{}}/>
      </MockedProvider>
    )
    expect(component).toMatchSnapshot();
  });


