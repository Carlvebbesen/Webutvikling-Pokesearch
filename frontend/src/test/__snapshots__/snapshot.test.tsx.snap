// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for App component 1`] = `ShallowWrapper {}`;

exports[`Snapshot test for Backdrop component 1`] = `ShallowWrapper {}`;

exports[`Snapshot test for Filter component 1`] = `ShallowWrapper {}`;

exports[`Snapshot test for ListPokemonRow component 1`] = `ShallowWrapper {}`;

exports[`Snapshot test for PopUp component 1`] = `ShallowWrapper {}`;

exports[`Snapshot test for SimpleTable component 1`] = `ShallowWrapper {}`;

exports[`Snapshot test for SortButton component 1`] = `ShallowWrapper {}`;

exports[`Snapshot test for SortList component 1`] = `ShallowWrapper {}`;
